components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -103,7 +115,145 @@ export default class PluginSettings extends React.Component {
         }
     }
 
+    handleActivate = async (pluginId) => {
+        this.setState({activating: pluginId});
+
+        const {error} = await this.props.actions.activatePlugin(pluginId);
+        this.setState({activating: null});
+
+        if (error) {
+            this.setState({serverError: error.message});
+        }
+    }
+
+    handleDeactivate = async (pluginId) => {
+        this.setState({deactivating: pluginId});
+
+        const {error} = await this.props.actions.deactivatePlugin(pluginId);
+        this.setState({deactivating: null});
+
+        if (error) {
+            this.setState({serverError: error.message});
+        }
+    }
+
+    buildPluginItem = (p) => {
+        let activateButton;
+        if (p.active) {
+            const deactivating = this.state.deactivating === p.id;
+            activateButton = (
+                <a
+                    disabled={deactivating}
+                    onClick={() => this.handleDeactivate(p.id)}
```

Comment:
is there a way you can remove the arrow function here and in the other places you do the same thing?

it seems that arrow functions or bindings affect performance as every time it renders it creates a new function

Reply:
I could probably do it but I'm pretty sure it's only a problem for anything that's a PureComponent or using shouldComponentUpdate, which this isn't